# DATAPRE Technical Report #

By Alon-alon, Cheng, Dela Cruz, Dizon, Evangelista

Sirs comment on our dataset:
<br>
Top Choice - If data will be coming from Kaggle, make sure you highlight how you applied preprocessing. Since data might be clean already, focus on feature engineering and descriptive analysis. Would also encourage the team to find a way to merge the analysis with other data sources if possible. Be creative with how you present the data and make sure that the question youre answering is clear. Suggestion on how you can answer the question in your top choice, check out ANOVA, regression, correlation and t tests :D Goodluck!!!

This technical report will explore the laptop market in order to get insights on its current state. Several questions will be tackled by tihs report: 
<br><br>
What are the common factors found among the current laptops in the market?
<br>
What are the common factors among the most sold laptops?

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In order to explore the common factors ... two laptop datasets were found

laptops_1.csv: https://www.kaggle.com/datasets/kuchhbhi/latest-laptop-price-list by Santosh Kumar
<br>
laptops_2.csv: https://www.kaggle.com/datasets/ionaskel/laptop-prices by Ionas Kelepouris   
<br>

## Convert the .csv files into dataframes ##

In [2]:
df_laptops1 = pd.read_csv("datasets/laptops_1.csv")

In [3]:
df_laptops1

brand     model processor_brand     processor_name processor_gnrtn  \
0     Lenovo   A6-9225             AMD  A6-9225 Processor            10th   
1     Lenovo   Ideapad             AMD           APU Dual            10th   
2      Avita      PURA             AMD           APU Dual            10th   
3      Avita      PURA             AMD           APU Dual            10th   
4      Avita      PURA             AMD           APU Dual            10th   
..       ...       ...             ...                ...             ...   
891     ASUS       ROG             AMD            Ryzen 9         Missing   
892     ASUS     Ryzen             AMD            Ryzen 9         Missing   
893     ASUS     Ryzen             AMD            Ryzen 9         Missing   
894  SAMSUNG    Galaxy        Qualcomm      Snapdragon 7c         Missing   
895   Lenovo  Thinkpad             AMD            Ryzen 5            10th   

      ram_gb ram_type      ssd      hdd       os  ... display_size  warranty  \
0    4 GB GB     DDR4     0 GB  1024 GB  Windows  ...      Missing         0   
1    4 GB GB     DDR4     0 GB   512 GB  Windows  ...      Missing         0   
2    4 GB GB     DDR4   128 GB     0 GB  Windows  ...      Missing         0   
3    4 GB GB     DDR4   128 GB     0 GB  Windows  ...      Missing         0   
4    4 GB GB     DDR4   256 GB     0 GB  Windows  ...      Missing         0   
..       ...      ...      ...      ...      ...  ...          ...       ...   
891  4 GB GB     DDR4  1024 GB     0 GB  Windows  ...           16         1   
892  4 GB GB     DDR4  1024 GB     0 GB  Windows  ...      Missing         1   
893  4 GB GB     DDR4  1024 GB     0 GB  Windows  ...           16         1   
894  4 GB GB  LPDDR4X     0 GB   512 GB  Windows  ...      Missing         0   
895  8 GB GB     DDR4   512 GB     0 GB      DOS  ...           14         0   

    Touchscreen msoffice  latest_price old_price discount  star_rating  \
0            No       No         24990     32790       23          3.7   
1            No       No         19590     21325        8          3.6   
2            No       No         19990     27990       28          3.7   
3            No       No         21490     27990       23          3.7   
4            No       No         24990     33490       25          3.7   
..          ...      ...           ...       ...      ...          ...   
891          No       No        144990    194990       25          0.0   
892          No       No        149990         0        0          0.0   
893          No       No        142990         0        0          0.0   
894          No       No         38990     47990       18          0.0   
895          No       No         57490     78400       26          4.2   

     ratings  reviews  
0         63       12  
1       1894      256  
2       1153      159  
3       1153      159  
4       1657      234  
..       ...      ...  
891        0        0  
892        0        0  
893        0        0  
894        0        0  
895       18        4  

[896 rows x 23 columns]

In [4]:
df_laptops2 = pd.read_csv("datasets/laptops_2.csv")

In [5]:
df_laptops2.head()

Unnamed: 0 Company      Product   TypeName  Inches  \
0           1   Apple  MacBook Pro  Ultrabook    13.3   
1           2   Apple  Macbook Air  Ultrabook    13.3   
2           3      HP       250 G6   Notebook    15.6   
3           4   Apple  MacBook Pro  Ultrabook    15.4   
4           5   Apple  MacBook Pro  Ultrabook    13.3   

                     ScreenResolution                         Cpu   Ram  \
0  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                            1440x900        Intel Core i5 1.8GHz   8GB   
2                   Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3  IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4  IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   

                Memory                           Gpu  OpSys  Weight  \
0            128GB SSD  Intel Iris Plus Graphics 640  macOS  1.37kg   
1  128GB Flash Storage        Intel HD Graphics 6000  macOS  1.34kg   
2            256GB SSD         Intel HD Graphics 620  No OS  1.86kg   
3            512GB SSD            AMD Radeon Pro 455  macOS  1.83kg   
4            256GB SSD  Intel Iris Plus Graphics 650  macOS  1.37kg   

   Price_euros  
0      1339.69  
1       898.94  
2       575.00  
3      2537.45  
4      1803.60

The first task is to merge the two laptop datasets together into one. But after loading the two different laptop datasets, it was noticed that they did not have exactly the same columns. There are some columns that are similar but named differently. But there are also some columns that exist in one dataset but not in the other.

With this in mind, it is needed to make the column names of the table uniform and decide which columns would be worth keeping for the new combined dataset.

## Columns of each laptop dataset ##

In [6]:
print("Columns of laptop dataset 1")
print(df_laptops1.columns)

Columns of laptop dataset 1
Index(['brand', 'model', 'processor_brand', 'processor_name',
       'processor_gnrtn', 'ram_gb', 'ram_type', 'ssd', 'hdd', 'os', 'os_bit',
       'graphic_card_gb', 'weight', 'display_size', 'warranty', 'Touchscreen',
       'msoffice', 'latest_price', 'old_price', 'discount', 'star_rating',
       'ratings', 'reviews'],
      dtype='object')


In [7]:
print("Columns of laptop dataset 2")
print(df_laptops2.columns)

Columns of laptop dataset 2
Index(['Unnamed: 0', 'Company', 'Product', 'TypeName', 'Inches',
       'ScreenResolution', 'Cpu', 'Ram', 'Memory', 'Gpu', 'OpSys', 'Weight',
       'Price_euros'],
      dtype='object')


The columns of the new laptop dataset are to be 

In [8]:
new_columns = ['brand', 'model', 'processor', 'ram', 'memory', 'os', 'display_size', 'price', 'star_rating', 'ratings']
df_laptops = pd.DataFrame(columns=new_columns)
df_laptops

Empty DataFrame
Columns: [brand, model, processor, ram, memory, os, display_size, price, star_rating, ratings]
Index: []

Each laptop dataset must first be cleaned in order for it to fit cohesively into this new dataset

# Initial dataset cleaning #

## Cleaning laptop dataset 1 ##

The first laptop dataset to be cleaned is currently stored in df_laptops1

In [9]:
df_laptops1

brand     model processor_brand     processor_name processor_gnrtn  \
0     Lenovo   A6-9225             AMD  A6-9225 Processor            10th   
1     Lenovo   Ideapad             AMD           APU Dual            10th   
2      Avita      PURA             AMD           APU Dual            10th   
3      Avita      PURA             AMD           APU Dual            10th   
4      Avita      PURA             AMD           APU Dual            10th   
..       ...       ...             ...                ...             ...   
891     ASUS       ROG             AMD            Ryzen 9         Missing   
892     ASUS     Ryzen             AMD            Ryzen 9         Missing   
893     ASUS     Ryzen             AMD            Ryzen 9         Missing   
894  SAMSUNG    Galaxy        Qualcomm      Snapdragon 7c         Missing   
895   Lenovo  Thinkpad             AMD            Ryzen 5            10th   

      ram_gb ram_type      ssd      hdd       os  ... display_size  warranty  \
0    4 GB GB     DDR4     0 GB  1024 GB  Windows  ...      Missing         0   
1    4 GB GB     DDR4     0 GB   512 GB  Windows  ...      Missing         0   
2    4 GB GB     DDR4   128 GB     0 GB  Windows  ...      Missing         0   
3    4 GB GB     DDR4   128 GB     0 GB  Windows  ...      Missing         0   
4    4 GB GB     DDR4   256 GB     0 GB  Windows  ...      Missing         0   
..       ...      ...      ...      ...      ...  ...          ...       ...   
891  4 GB GB     DDR4  1024 GB     0 GB  Windows  ...           16         1   
892  4 GB GB     DDR4  1024 GB     0 GB  Windows  ...      Missing         1   
893  4 GB GB     DDR4  1024 GB     0 GB  Windows  ...           16         1   
894  4 GB GB  LPDDR4X     0 GB   512 GB  Windows  ...      Missing         0   
895  8 GB GB     DDR4   512 GB     0 GB      DOS  ...           14         0   

    Touchscreen msoffice  latest_price old_price discount  star_rating  \
0            No       No         24990     32790       23          3.7   
1            No       No         19590     21325        8          3.6   
2            No       No         19990     27990       28          3.7   
3            No       No         21490     27990       23          3.7   
4            No       No         24990     33490       25          3.7   
..          ...      ...           ...       ...      ...          ...   
891          No       No        144990    194990       25          0.0   
892          No       No        149990         0        0          0.0   
893          No       No        142990         0        0          0.0   
894          No       No         38990     47990       18          0.0   
895          No       No         57490     78400       26          4.2   

     ratings  reviews  
0         63       12  
1       1894      256  
2       1153      159  
3       1153      159  
4       1657      234  
..       ...      ...  
891        0        0  
892        0        0  
893        0        0  
894        0        0  
895       18        4  

[896 rows x 23 columns]

In the new dataset, processor_brand and processor_name is combined into one column

In [10]:
df_laptops1['processor'] = df_laptops1['processor_brand'] + ' ' + df_laptops1['processor_name']
df_laptops1['processor']

0       AMD A6-9225 Processor
1                AMD APU Dual
2                AMD APU Dual
3                AMD APU Dual
4                AMD APU Dual
                ...          
891               AMD Ryzen 9
892               AMD Ryzen 9
893               AMD Ryzen 9
894    Qualcomm Snapdragon 7c
895               AMD Ryzen 5
Name: processor, Length: 896, dtype: object

The ram column was also noticed to be incorrectly formatted, with GB being mentioned twice for all rows

In [11]:
df_laptops1['ram_gb']

0      4 GB GB
1      4 GB GB
2      4 GB GB
3      4 GB GB
4      4 GB GB
        ...   
891    4 GB GB
892    4 GB GB
893    4 GB GB
894    4 GB GB
895    8 GB GB
Name: ram_gb, Length: 896, dtype: object

By splitting each value in the ram column and keeping the first two words, the format would now be aligned with the second dataset

In [12]:
df_laptops1['ram'] = df_laptops1['ram_gb'].apply(lambda x: x.split()[0] + x.split()[1])
df_laptops1['ram']

0      4GB
1      4GB
2      4GB
3      4GB
4      4GB
      ... 
891    4GB
892    4GB
893    4GB
894    4GB
895    8GB
Name: ram, Length: 896, dtype: object

ssd and hdd columns will also be combined into a column named memory

In [13]:
df_laptops1.loc[df_laptops1['hdd'] == '0 GB', ['memory']] = df_laptops1['ssd'] + ' SSD'
df_laptops1.loc[df_laptops1['ssd'] == '0 GB', ['memory']] = df_laptops1['hdd'] + ' HDD'
df_laptops1.loc[(df_laptops1['ssd'] != '0 GB') & (df_laptops1['hdd'] != '0 GB'), ['memory']] = df_laptops1['ssd'] + ' SSD + ' + df_laptops1['hdd'] + ' HDD'

In [14]:
df_laptops1['memory'] = df_laptops1['memory'].str.replace(' GB', 'GB')

In [15]:
df_laptops1.memory.value_counts()

512GB SSD                 412
256GB SSD                 128
1024GB SSD                111
1024GB HDD                 87
256GB SSD + 1024GB HDD     73
512GB HDD                  63
128GB SSD                  11
512GB SSD + 1024GB HDD      3
2048GB SSD                  2
512GB SSD + 512GB HDD       2
3072GB SSD                  1
2048GB HDD                  1
128GB SSD + 1024GB HDD      1
32GB SSD                    1
Name: memory, dtype: int64

As the laptop dataset 1 has two price columns (old price and latest price), only the latest price will be kept as this report tries to explore the current state of the laptop market

In [16]:
df_laptops1['price'] = df_laptops1['latest_price']

Also, it was noticed that the currency of the two laptop datasets are different. So in order to make it uniform the currency is to be converted in to USD.

For laptop dataset 1, the prices are in Indian Rupee (INR).

In [17]:
INR_to_USD = 0.013
df_laptops1['price'] = df_laptops1['price'] * INR_to_USD

## Append the cleaned laptop 1 columns to the new dataframe ##

In [18]:
df_laptops1_cols = df_laptops1[['brand', 'model', 'processor', 'ram', 'memory', 'os',
                                'display_size', 'price', 'star_rating', 'ratings']]

df_laptops = df_laptops.append(df_laptops1_cols)
df_laptops

brand     model               processor  ram      memory       os  \
0     Lenovo   A6-9225   AMD A6-9225 Processor  4GB  1024GB HDD  Windows   
1     Lenovo   Ideapad            AMD APU Dual  4GB   512GB HDD  Windows   
2      Avita      PURA            AMD APU Dual  4GB   128GB SSD  Windows   
3      Avita      PURA            AMD APU Dual  4GB   128GB SSD  Windows   
4      Avita      PURA            AMD APU Dual  4GB   256GB SSD  Windows   
..       ...       ...                     ...  ...         ...      ...   
891     ASUS       ROG             AMD Ryzen 9  4GB  1024GB SSD  Windows   
892     ASUS     Ryzen             AMD Ryzen 9  4GB  1024GB SSD  Windows   
893     ASUS     Ryzen             AMD Ryzen 9  4GB  1024GB SSD  Windows   
894  SAMSUNG    Galaxy  Qualcomm Snapdragon 7c  4GB   512GB HDD  Windows   
895   Lenovo  Thinkpad             AMD Ryzen 5  8GB   512GB SSD      DOS   

    display_size    price  star_rating ratings  
0        Missing   324.87          3.7      63  
1        Missing   254.67          3.6    1894  
2        Missing   259.87          3.7    1153  
3        Missing   279.37          3.7    1153  
4        Missing   324.87          3.7    1657  
..           ...      ...          ...     ...  
891           16  1884.87          0.0       0  
892      Missing  1949.87          0.0       0  
893           16  1858.87          0.0       0  
894      Missing   506.87          0.0       0  
895           14   747.37          4.2      18  

[896 rows x 10 columns]

## Cleaning laptop dataset 2 ##

The second laptop dataset to be cleaned is stored in df_laptops2

In [19]:
df_laptops2

Unnamed: 0 Company                              Product  \
0              1   Apple                          MacBook Pro   
1              2   Apple                          Macbook Air   
2              3      HP                               250 G6   
3              4   Apple                          MacBook Pro   
4              5   Apple                          MacBook Pro   
...          ...     ...                                  ...   
1298        1316  Lenovo                       Yoga 500-14ISK   
1299        1317  Lenovo                       Yoga 900-13ISK   
1300        1318  Lenovo                   IdeaPad 100S-14IBR   
1301        1319      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon   
1302        1320    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)   

                TypeName  Inches                            ScreenResolution  \
0              Ultrabook    13.3          IPS Panel Retina Display 2560x1600   
1              Ultrabook    13.3                                    1440x900   
2               Notebook    15.6                           Full HD 1920x1080   
3              Ultrabook    15.4          IPS Panel Retina Display 2880x1800   
4              Ultrabook    13.3          IPS Panel Retina Display 2560x1600   
...                  ...     ...                                         ...   
1298  2 in 1 Convertible    14.0   IPS Panel Full HD / Touchscreen 1920x1080   
1299  2 in 1 Convertible    13.3  IPS Panel Quad HD+ / Touchscreen 3200x1800   
1300            Notebook    14.0                                    1366x768   
1301            Notebook    15.6                                    1366x768   
1302            Notebook    15.6                                    1366x768   

                                       Cpu   Ram               Memory  \
0                     Intel Core i5 2.3GHz   8GB            128GB SSD   
1                     Intel Core i5 1.8GHz   8GB  128GB Flash Storage   
2               Intel Core i5 7200U 2.5GHz   8GB            256GB SSD   
3                     Intel Core i7 2.7GHz  16GB            512GB SSD   
4                     Intel Core i5 3.1GHz   8GB            256GB SSD   
...                                    ...   ...                  ...   
1298            Intel Core i7 6500U 2.5GHz   4GB            128GB SSD   
1299            Intel Core i7 6500U 2.5GHz  16GB            512GB SSD   
1300  Intel Celeron Dual Core N3050 1.6GHz   2GB   64GB Flash Storage   
1301            Intel Core i7 6500U 2.5GHz   6GB              1TB HDD   
1302  Intel Celeron Dual Core N3050 1.6GHz   4GB            500GB HDD   

                               Gpu       OpSys  Weight  Price_euros  
0     Intel Iris Plus Graphics 640       macOS  1.37kg      1339.69  
1           Intel HD Graphics 6000       macOS  1.34kg       898.94  
2            Intel HD Graphics 620       No OS  1.86kg       575.00  
3               AMD Radeon Pro 455       macOS  1.83kg      2537.45  
4     Intel Iris Plus Graphics 650       macOS  1.37kg      1803.60  
...                            ...         ...     ...          ...  
1298         Intel HD Graphics 520  Windows 10   1.8kg       638.00  
1299         Intel HD Graphics 520  Windows 10   1.3kg      1499.00  
1300             Intel HD Graphics  Windows 10   1.5kg       229.00  
1301            AMD Radeon R5 M330  Windows 10  2.19kg       764.00  
1302             Intel HD Graphics  Windows 10   2.2kg       369.00  

[1303 rows x 13 columns]

Firstly, the columns should be renamed in order to be the same with the new dataset

In [20]:
df_laptops2.rename(columns = {'Company':'brand', 'Product':'model', 'Inches':'display_size', 'Cpu':'processor',
                              'Price_euros':'price', 'OpSys':'os', 'Ram':'ram', 'Memory':'memory'}, inplace = True)

It was noticed that in the memory column, there were many excess spaces so this had to be fixed

In [21]:
df_laptops2['memory'].value_counts()

256GB SSD                        412
1TB HDD                          223
500GB HDD                        132
512GB SSD                        118
128GB SSD +  1TB HDD              94
128GB SSD                         76
256GB SSD +  1TB HDD              73
32GB Flash Storage                38
2TB HDD                           16
64GB Flash Storage                15
512GB SSD +  1TB HDD              14
1TB SSD                           14
256GB SSD +  2TB HDD              10
1.0TB Hybrid                       9
256GB Flash Storage                8
16GB Flash Storage                 7
32GB SSD                           6
180GB SSD                          5
128GB Flash Storage                4
16GB SSD                           3
512GB SSD +  2TB HDD               3
256GB SSD +  256GB SSD             2
256GB SSD +  500GB HDD             2
512GB Flash Storage                2
1TB SSD +  1TB HDD                 2
128GB SSD +  2TB HDD               2
128GB HDD                          1
5

In [22]:
df_laptops2['memory'] = df_laptops2['memory'].apply(lambda x: ' '.join(x.split()))

In [23]:
df_laptops2['memory'].value_counts()

256GB SSD                       412
1TB HDD                         223
500GB HDD                       132
512GB SSD                       118
128GB SSD + 1TB HDD              94
128GB SSD                        76
256GB SSD + 1TB HDD              73
32GB Flash Storage               38
2TB HDD                          16
64GB Flash Storage               15
512GB SSD + 1TB HDD              14
1TB SSD                          14
256GB SSD + 2TB HDD              10
1.0TB Hybrid                      9
256GB Flash Storage               8
16GB Flash Storage                7
32GB SSD                          6
180GB SSD                         5
128GB Flash Storage               4
16GB SSD                          3
512GB SSD + 2TB HDD               3
256GB SSD + 256GB SSD             2
128GB SSD + 2TB HDD               2
512GB Flash Storage               2
1TB SSD + 1TB HDD                 2
256GB SSD + 500GB HDD             2
512GB SSD + 256GB SSD             1
64GB Flash Storage + 1TB HDD

Just like with the prior dataset, the price column must be converted in the a uniform currency for the new dataset (USD).

For laptop dataset 2, the prices are in euro.

In [24]:
EUR_to_USD = 1.06
df_laptops2['price'] = df_laptops2['price'] * EUR_to_USD

The new dataset will also contain the ratings and star_rating column, however as this laptop dataset does not originally contain this, values will be initially set to zero

In [25]:
df_laptops2['star_rating'] = 0.0
df_laptops2['ratings'] = 0

## Append the cleaned laptop 2 columns to the new dataframe ##

In [26]:
df_laptops2_cols = df_laptops2[['brand', 'model', 'processor', 'ram', 'memory', 'os',
                                'display_size', 'price', 'star_rating', 'ratings']]

df_laptops = df_laptops.append(df_laptops2_cols)
df_laptops

brand                                model  \
0     Lenovo                              A6-9225   
1     Lenovo                              Ideapad   
2      Avita                                 PURA   
3      Avita                                 PURA   
4      Avita                                 PURA   
...      ...                                  ...   
1298  Lenovo                       Yoga 500-14ISK   
1299  Lenovo                       Yoga 900-13ISK   
1300  Lenovo                   IdeaPad 100S-14IBR   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)   

                                 processor   ram              memory  \
0                    AMD A6-9225 Processor   4GB          1024GB HDD   
1                             AMD APU Dual   4GB           512GB HDD   
2                             AMD APU Dual   4GB           128GB SSD   
3                             AMD APU Dual   4GB           128GB SSD   
4                             AMD APU Dual   4GB           256GB SSD   
...                                    ...   ...                 ...   
1298            Intel Core i7 6500U 2.5GHz   4GB           128GB SSD   
1299            Intel Core i7 6500U 2.5GHz  16GB           512GB SSD   
1300  Intel Celeron Dual Core N3050 1.6GHz   2GB  64GB Flash Storage   
1301            Intel Core i7 6500U 2.5GHz   6GB             1TB HDD   
1302  Intel Celeron Dual Core N3050 1.6GHz   4GB           500GB HDD   

              os display_size    price  star_rating ratings  
0        Windows      Missing   324.87          3.7      63  
1        Windows      Missing   254.67          3.6    1894  
2        Windows      Missing   259.87          3.7    1153  
3        Windows      Missing   279.37          3.7    1153  
4        Windows      Missing   324.87          3.7    1657  
...          ...          ...      ...          ...     ...  
1298  Windows 10         14.0   676.28          0.0       0  
1299  Windows 10         13.3  1588.94          0.0       0  
1300  Windows 10         14.0   242.74          0.0       0  
1301  Windows 10         15.6   809.84          0.0       0  
1302  Windows 10         15.6   391.14          0.0       0  

[2199 rows x 10 columns]

## Now the two datasets have been combined! ##

With a new total of 2199 rows, coming from 896 records in the first dataset and 1303 records in the second dataset

In [27]:
df_laptops.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2199 entries, 0 to 1302
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   brand         2199 non-null   object 
 1   model         2199 non-null   object 
 2   processor     2199 non-null   object 
 3   ram           2199 non-null   object 
 4   memory        2199 non-null   object 
 5   os            2199 non-null   object 
 6   display_size  2199 non-null   object 
 7   price         2199 non-null   float64
 8   star_rating   2199 non-null   float64
 9   ratings       2199 non-null   object 
dtypes: float64(2), object(8)
memory usage: 189.0+ KB


## More cleaning ... ##

Todo: double check for inconsistent data

In [28]:
df_laptops.brand.value_counts()

Lenovo       445
HP           416
Dell         297
ASUS         254
Asus         158
DELL         154
MSI          106
Acer         103
acer          58
Toshiba       48
APPLE         28
Apple         21
Avita         18
Samsung        9
LG             8
Razer          7
Mediacom       7
Microsoft      6
Vaio           5
Nokia          4
Xiaomi         4
realme         4
Infinix        4
Vero           4
ALIENWARE      4
Fujitsu        3
Chuwi          3
Smartron       3
lenovo         3
RedmiBook      3
Google         3
MICROSOFT      3
Huawei         2
Mi             2
SAMSUNG        1
iball          1
Name: brand, dtype: int64